src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -330,6 +339,58 @@ applyLocal logger dbEnv gasModel pd spv cmdIn mc =
 
       checkTooBigTx gas0 gasLimit (applyPayload em) return
 
+applyUpgrades
+  :: ChainwebVersion
+  -> BlockHeader
+  -> BlockCreationTime
+  -> TransactionM p (Maybe ModuleCache)
+applyUpgrades v parentHeader (BlockCreationTime currCreationTime) =
+  case upgradeCoinV2Date v of
+    Nothing -> testBlock1
+    Just t -> testUpgradeTime t
+  where
+
+    testBlock1 | parentBlockHeight == 0 = go
+               | otherwise = noop
+    parentBlockHeight = _blockHeight parentHeader
+
+    testUpgradeTime t | isUpgradeBlockTime t = go
+                      | otherwise = noop
+    isUpgradeBlockTime upgradeTime =
+      parentTime < upgradeTime
+      &&
+      upgradeTime <= currCreationTime
```

Comment:
Aren't we guaranteed that `parentTime < currCreationTime`? If so, then shouldn't we just test that `upgradeTime <= currCreationTime`?  Also, I don't believe `runCoinbase` is called for genesis transactions, so we won't need to check against whether a particular parent header was at genesis as it's implied.

Reply:
@larskuhtz can you weigh in here? This is your proposed logic (as implemented by me)

Reply:
> If so, then shouldn't we just test that upgradeTime <= currCreationTime? 
We need just one block, not every block ever before currCreationTime

Reply:
@emilypi I don't understand the second comment, which seems to refer to `PactService`

Reply:
@slpopejoy It is referring to `PactService`. `runCoinbase` in `PactService` will not run coinbases for genesis blocks, so every block that hits `applyCoinbase` will by definition be a non-genesis block. So rather than `parentBlockHeight == 0`, really what we're looking for is `currBlockHeight == 1`. 

I'm trying to make the argument that passing in a parent header here is not really going to contribute anything.